misk-hibernate/src/main/kotlin/misk/hibernate/SessionFactoryService.kt

```diff
@@ -104,19 +111,73 @@ internal class SessionFactoryService(
 
     val registry = registryBuilder.build()
 
+    // Register entity types.
     val metadataSources = MetadataSources(registry)
     for (entityClass in entityClasses) {
       metadataSources.addAnnotatedClass(entityClass.entity.java)
     }
-    val metadataBuilder = metadataSources.metadataBuilder
-    metadataBuilder.applyBasicType(IdType, Id::class.qualifiedName)
-    metadataBuilder.applyBasicType(ByteStringType, ByteString::class.qualifiedName)
-    val metadata = metadataBuilder.build()
+
+    val metadataBuilder: MetadataBuilder = metadataSources.metadataBuilder
```

Comment:
nit: the metadataBuilder variable is no longer used (aside from calling build() immediately)

Reply:
Added a comment to note how scary Hibernate is!